lib/model/query/analytics.js

```diff
@@ -603,13 +603,18 @@ const previewMetrics = () => (({ Analytics }) => Promise.all([
   return metrics;
 }));
 
-// Get most recent 'analytics' audit from the past N days
-const AUDIT_SCHEDULE = config.has('default.taskSchedule.analytics')
+// Usage reports are sent on a fixed interval: a usage report is sent a fixed
+// number of days after the previous report. The default is to send a report
+// every 31 days. However, that number is configurable so that usage reporting
+// can be verified during regression testing.
+const ANALYTICS_SCHEDULE = config.has('default.taskSchedule.analytics')
   ? config.get('default.taskSchedule.analytics')
-  : 30; // Default is 30 days
+  : 31; // Default is 31 days
 
+// Returns the latest recent attempt to send a usage report. If there is no
+// recent attempt, then it is time for a new report to be sent.
 const getLatestAudit = () => ({ maybeOne }) => maybeOne(sql`select * from audits
-  where action='analytics' and "loggedAt" >= current_date - cast(${AUDIT_SCHEDULE} as int)
+  where action='analytics' and current_date - "loggedAt"::date < ${ANALYTICS_SCHEDULE}
```

Comment:
Instead of this change, I could have done:

```sql
"loggedAt" >= current_date - cast(${AUDIT_SCHEDULE} as int) + 1
```

However, I think that would have been less clear overall.

I was initially confused about dates vs. timestamps here. Previously, `current_date - cast(${AUDIT_SCHEDULE} as int)` was implicitly coerced to a timestamp with time part equal to midnight. I started wondering about the time part of `loggedAt` and whether it mattered that it was 3 a.m. instead of midnight. Now I'm converting `loggedAt` to a date and calculating the number of days between that date and `current_date`. That way, the time part of `loggedAt` doesn't figure into things: it doesn't matter when the usage report started or how long it took to complete.

This approach also lets us remove the cast of `ANALYTICS_SCHEDULE`.

Reply:
I like this new approach a lot of making both values dates (without timestamps) and doing the subtraction there.